docs/plus/pvc.md

```diff
@@ -0,0 +1,77 @@
+# PersistentVolumeClaim
+
+A `PersistentVolumeClaim` (PVC) is a request for storage by a pod.
+
+!!! tip ""
+    [API Reference](../reference/cdk8s-plus-22/typescript.md#persistent-volume-claim)
+
+A `PersistentVolumeClaim` contains the requirements of the request, and the Kubernetes control plane is responsible providing a physical volume that satisfies the claim's requirements.
+
+```ts
+import * as kplus from 'cdk8s-plus-22';
+import * as cdk8s from 'cdk8s';
+
+const pod = new kplus.Pod(chart, 'Pod');
+const container = pod.addContainer({ image: 'node' });
+
+// create the storage request
+const claim = new kplus.PersistentVolumeClaim(chart, 'Claim', {
+  storage: cdk8s.Size.gibibytes(50),
+});
```

Comment:
How do we envision supporting CSI-based [PVC cloning](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#create-persistent-volume-claim-from-an-existing-pvc) and [PVC from Volume Snapshot](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#volume-snapshot-and-restore-volume-from-snapshot-support)?

How about something like this?

```ts
const claimFromClone = new kplus.PersistentVolumeClaim(chart, 'Claim', {
  dataSource: PersistentVolumeClaimDataSource.pvcDataSource(otherClaim),
});

const claimFromSnapshot = new kplus.PersistentVolumeClaim(chart, 'Claim', {
  dataSource: PersistentVolumeClaimDataSource.snapshotDataSource({
    name: 'new-snapshot-test',
    kind: 'VolumeSnapshot',
    apiGroup: 'snapshot.storage.k8s.io',
  }),
});
```

Reply:
@misterjoshua yeah I think you're proposed API is the right direction. We will create a union like class called `PersistentVolumeClaimDataSource` that will offer all these data sources 👍 

We can add it as an enhancement later on, do you wanna have a go at adding it? :)